roxctl/netpol/connectivity/diff/netpol_test.go

```diff
@@ -0,0 +1,249 @@
+package diff
+
+import (
+	"errors"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/roxctl/common/environment/mocks"
+	"github.com/stackrox/rox/roxctl/common/npg"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestDiffAnalyzeNetpolCommand(t *testing.T) {
+	t.Parallel()
+	suite.Run(t, new(diffAnalyzeNetpolTestSuite))
+}
+
+type diffAnalyzeNetpolTestSuite struct {
+	suite.Suite
+}
+
+func (d *diffAnalyzeNetpolTestSuite) TestDiffAnalyzeNetpol() {
+	tmpOutFileName := d.T().TempDir() + "/out"
+	outFileTxt := tmpOutFileName + ".txt"
+	cases := []struct {
+		name                  string
+		inputFolderPath1      string
+		inputFolderPath2      string
+		strict                bool
+		stopOnFirstErr        bool
+		outFile               string
+		outputToFile          bool
+		outputFormat          string
+		removeOutputPath      bool
+		errStringContainment  bool
+		expectedValidateError error
+		expectedAnalysisError error
+	}{
+		{
+			name:                  "Empty dir1 input should return validation error",
+			inputFolderPath1:      "",
+			inputFolderPath2:      "",
+			expectedValidateError: errox.InvalidArgs,
+		},
+		{
+			name:                  "Empty dir2 should return validation error",
+			inputFolderPath1:      "/dev/null",
+			inputFolderPath2:      "",
+			expectedValidateError: errox.InvalidArgs,
+		},
+		{
+			name:                  "Not existing input folder paths should result in error 'os.ErrNotExist'",
+			inputFolderPath1:      "/tmp/xxx",
+			inputFolderPath2:      "/tmp/xxx",
+			expectedValidateError: nil,
+			expectedAnalysisError: os.ErrNotExist,
+		},
```

Comment:
```
$  bin/darwin_amd64/roxctl netpol connectivity diff --dir1 /tmp --dir2 /nonexisting
ERROR:  error in connectivity diff analysis: error accessing directory: lstat /nonexisting: no such file or directory
```

This is a pretty raw error. Could we wrap it in errox or npg.Error (or maybe both?).

Ideally the error message should communicate that `dir2` was not found.